### PR TITLE
resgroup: add test cases for name convention.

### PR DESCRIPTION
**Pull request by pivotal-ning-yu**
resgroup name convention:
- min length is 1 character;
- max length is 63 characters;
- characters exceed the max length are ignored;
- ~~case insensitive;~~
- case sensitive;
- can be around with quotation marks `""`;
- without quotation marks:
  - always converted to lower case;
  - can not begin with numbers;
  - punctuations are not allowed;
  - white spaces are not allowed;
- with quotation marks:
  - can begin with numbers;
  - punctuations are allowed;
  - white spaces are allowed;
- CJK characters are supported w/ or w/o quotation marks;
- Unicode escapes are supported (e.g. `U&"\0072\0067"`);
- backslash escapes (e.g. `\\`, `\t`, `\x31`, etc.) are not supported;

### CODE REVIEW

**Comment by pivotal-ning-yu** [comment]
A new commit was submitted to also test on ALTER command and the resgroup views.

**Comment by pivotal-ning-yu** [comment]
Yes it's put under isolation2 just because other resgroup tests are put
there. Most of resgroup tests requires the isolation2 test framework, but
this one not. It makes sense to add a comment here. I'll update it soon.
Thanks for the suggestion.

Also to mention that although resgroup tests are put under isolation2 dir
they are not executed together with the isolation2 tests, so `make -C
src/test/isolation2 installcheck` is just the original isolation2 test, and
`make installcheck-resgroup` is the resgroup test. We make it this way
because we need to execute resgroup test repeatedly and quickly during our
development. We may move resgroup tests into isolation2 and/or ICW sooner
or later.

On Sat, Oct 14, 2017 at 2:51 AM, Daniel Gustafsson <notifications@github.com

File: src/test/isolation2/sql/resgroup/resgroup_name_convention.sql [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+-- Resource group names follow the general object name convention:
+-- https://www.postgresql.org/docs/current/static/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
+--
+-- Besides that there are 3 reserved names:
+-- * default_group, admin_group: names of the 2 default resource groups;
+-- * none: a special name usually used in ALTER ROLE command to reset
+--   to the proper default resource group;
+--
+-- This case is put under isolation2 dir as other resource group cases,
+-- although it does not requires any extended feature of the isolation2
```
Comment (xiong-gang) [inline src/test/isolation2/sql/resgroup/resgroup_name_convention.sql]: s/requires/require

**Comment by pivotal-ning-yu** [comment]
Thanks for the review, will merge now.

**Pull request merged**